### DISCUSSION TITLE
Membership management for Blazorise apps

### DISCUSSION THREAD

**Discussion by reubenalfred** [Megabit/Blazorise#4827] [Q&A] [answered]
Greetings my super smart Blazorise community!!

As I mentioned in Discord, sorry, I will stick to this forum in the future, what would be the best strategy to include membership management for Blazorise apps?

Obviously, the VS scaffolder cannot work with Blazorise template-generated projects without some modifications.

**Comment by David-Moreira** [comment]
Hello @reubenalfred 
We have plans to provide an option for Authentication/Authorization in the templates in the future (My guess is that this is what you're looking for?). But unfortunately we haven't gotten around to it just yet.

My suggestion is that you pick a template that already has this management as that's more complex to implement and then add `Blazorise` by either
- downloading one of our templates and copy pasting the bits that matter or 
- following the installation instructions for your provider of choice: https://blazorise.com/docs/usage

**Reply by reubenalfred** [reply]
Thank you so much, @David-Moreira very much appreciated mate, yes it is what I meant and yes the security aspect of it is better to be left to well-written code!

**Reply by reubenalfred** [reply]
>> We have plans to provide an option for Authentication/Authorization in the templates in the future (My guess is that this is what you're looking for?). But unfortunately we haven't gotten around to it just yet.

Cannot wait for that @David-Moreira, it will be well worth the wait!! :)

**Reply by reubenalfred** [reply]
Hi again @David-Moreira , sorry to be a jab-in-the-bumbo, just to make sure that I understood your instructions

>> downloading one of our templates and copy-pasting the bits that matter or

did you mean the project templates for VS or some other Blazor project on GitHub with Blazorise and .NET Core Identity integrated into it, if so where would I find this project on GitHub?

>> following the installation instructions for your provider of choice: https://blazorise.com/docs/usage

Did you mean provider as in membership provider or the style provider?


FYI,

I have the Blazorise template extension for VS 2022 installed already and used that to generate my project.

**Reply by David-Moreira** [reply, marked as answer]
Unfortunately I don't know of any Blazorise project with .NET Core Identity integrated into it. Anyway it's really easy to add Blazorise onto an existing project, as you'll find out by visiting the usage page for the provider of your liking (bootstrap, bulma, material, etc...).

I would recommend going with Microsoft's project template with inbuilt auth and then add Blazorise on top since you just need to follow the steps from https://blazorise.com/docs/usage (by choosing the css framework provider of your liking)

Now depending on your use case, and if you're new to .NET Core Identity it can be confusing to configure/customize/extend the functionalities, so I would recommend looking up some tutorial or a better bootstrapped template. 
And also if I remember correctly, nowadays the newer templates come with a different flavor of identity, Duende's Identity Server (which needs a license for commercial apps), so you might have to do some investigation there. :) Good luck just let us know if you need any help `Blazorise` related.

**Reply by stsrki** [reply]
The Identity stuff is quite hard to set up if you go too far from Microsoft's project template with inbuilt auth. We did a lot of work to make it on our projects. I advise doing it as David said: go with the default Microsoft project templates and then adjust where necessary.

PS. As part of the Blazorise Enterprise license, we give access to the entire source code for https://blazorise.com/support, which has a fully customized Identity authentication.

**Reply by reubenalfred** [reply]
Thank you very much @David-Moreira And @stsrki!

I really didn't think I'd get a reply this soon!!

Much obliged!

As for the suggestion made by @David-Moreira, that's exactly what I started doing but I realised it was a big task so thought I'd ask the gurus first.

I will keep you posted guys!

 As for the pro license the price is very generous I will need AUD750 so I am thinking of saving up for 3 months but in the mean time will do what you guys suggested!

**Comment by reubenalfred** [comment]
The project i am working on is to help myself master blazor and Blazorise. I have tried almost all the Blazor component libraries on the market other than Telerik's and DevCrafts and boy o boy Blazorise is sweet.

I have been a desktop developer for over 25 years and now looking in to web dev so Blazor is the one I chose.

My project is called SkripKwiz it is a web app for friends to test improve the scriptural knowledge.

Part of the project is to build a generic theme manager to reuse with Blazorise projects. The idea is to support color schemes and geometric aspects of styling but will start with color styling first.

Linear, Rectangular, Elliptic and Text entities. each are colorable and have various stylable aspects. most of the controls can be defined in terms of the fundamental entities mentioned above. For example, a LIstBox has a header, panel, scrollbars(rail and thumbnail), item canvas, 4 corners with radii. all the primitives can be themed using the 4 entities mentioned. furthermore, each control styles encapsulated can have states too and styles for each state can be stored in a dictionary using the same object to abstract the styles for a given control.

we have the classes to describe the styles then we describe the actual scheme. we specify different states: default, hover, disabled, focused, selected and so on. the colors for these various states are expressed in terms % color attributes with respect to the default. borders too can be computed relative to the background color derived from the base color. this can be applied to any visual entity.

The base color and other colors in default can be directly chosen too as opposed to algorithmic based on the base color.

once the theme is chosen(obviously color choosers and slider will be made available, also a control for selecting various colo systems) will be exported as json file. then from this json Blazorise specific CSS can be generated,

well, that's idea..... :)

**Reply by reubenalfred** [reply]
this is the one i was working on sometime back

![brazil](https://github.com/Megabit/Blazorise/assets/115092608/edb678aa-abf8-4aee-bf08-d5d0ba5aac4b)

**Reply by stsrki** [reply]
Wow, that is one complex theme editor 😅

**Comment by reubenalfred** [comment]
Thank you @David-Moreira and @stsrki I have followed the suggested strategy and fixed all the build errors and so far so good but the styling has gone haywire compared to the styling from the project generated by the Blazorise proj template...

CSS is my worst weakness :)

please note the marked concerns in one of the screenshots...

![skrip-kwiz-material](https://github.com/Megabit/Blazorise/assets/115092608/83b6b169-1799-4c25-863c-05fa0e627e19)

![skrip-kwiz-blazorise](https://github.com/Megabit/Blazorise/assets/115092608/e4de1bea-33af-4400-8b1d-060827f34c7e)

**Comment by reubenalfred** [comment]
OK, all good, the material static files were not present under the static directory, so cosmetically the new project looks good..

now back to Identity docos :)

thanks guys!!
